apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx

```diff
@@ -0,0 +1,199 @@
+        mode: 'onChange'
+    });
+    const [selectedFieldType, setSelectedFieldType] = useState('');
+    const { pageId, ruleId } = useParams();
+    const { showAlert } = useAlert();
+
+    useEffect(() => {
+        if (ruleId) {
+            PageRuleControllerService.viewRuleResponseUsingGet({
+                authorization: authorization(),
+                ruleId: Number(ruleId)
+            }).then((resp: ViewRuleResponse) => {
+                form.setValue('anySourceValue', resp?.anySourceValue!);
+                form.setValue('ruleDescription', resp?.ruleDescription!);
+                form.setValue('ruleFunction', resp?.ruleFunction!);
+                form.setValue('comparator', resp?.comparator!);
+                form.setValue('targetType', resp?.targetType!);
+                form.setValue('sourceIdentifier', resp?.sourceIdentifier!);
+                form.setValue('sourceText', resp?.sourceIdentifier!);
+                form.setValue('sourceValueIds', resp?.sourceValue?.sourceValueId!);
+                form.setValue('sourceValueText', resp?.sourceValue?.sourceValueText!);
+                form.setValue('targetValueIdentifier', resp?.targetValueIdentifier!);
+                form.setValue('targetValueText', ['testing']);
+                setSelectedFieldType(resp.ruleFunction!);
+            });
+        }
+    }, [ruleId]);
+
+    const onSubmit = form.handleSubmit(async (data) => {
+        const sourceValue = {
+            sourceValueId: data.sourceValueIds,
+            sourceValueText: data.sourceValueText
+        };
+        const request = {
+            anySourceValue: data.anySourceValue,
+            comparator: data.comparator,
+            ruleDescription: data.ruleDescription,
+            ruleFunction: data.ruleFunction,
+            sourceIdentifier: data.sourceIdentifier,
+            sourceText: data.sourceText,
+            sourceValue: sourceValue,
+            targetType: data.targetType,
+            targetValueIdentifier: data.targetValueIdentifier,
+            targetValueText: data.targetValueText
+        };
+        if (!ruleId) {
+            PageRuleControllerService.createBusinessRuleUsingPost({
+                authorization: authorization(),
+                id: Number(pageId),
+                request
+            }).then((resp) => {
+                console.log('resp', resp);
+                showAlert({ type: 'success', header: 'added', message: resp.message });
+            });
+        } else {
+            PageRuleControllerService.updatePageRuleUsingPut({
+                authorization: authorization(),
+                page: Number(pageId),
+                ruleId: Number(ruleId),
+                request
+            }).then((resp) => {
+                showAlert({ type: 'success', header: 'updated', message: resp.message });
+            });
+        }
+        handleCancel();
+    });
+
+    const handleCancel = () => {
+        navigate(-1);
+    };
```

Comment:
Issue: `navigate(-1)` can cause some weird behavior and should be avoided if possible. For example, what if a user bookmarks (for some weird reason) the add rule page from somewhere unexpected. 

Alternative without looking at whats actually going on:

if the route to add a new rule is something like
` page-builder/pages/1234/rules/add`
then using `navigate('..')` would send them to `page-builder/pages/1234/rules`